idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1036,6 +1054,24 @@ struct GetWorkflowExecutionHistoryResponse {
   20: optional binary nextPageToken
 }
 
+struct GetWorkflowExecutionRawHistoryRequest {
+  10: optional string domain
+  20: optional WorkflowExecution execution
+  30: optional binary branchToken
+  40: optional i64 (js.type = "Long") firstEventId
+  50: optional i64 (js.type = "Long") nextEventId
+  60: optional i32 maximumPageSize
+  70: optional binary nextPageToken
+}
+
+struct GetWorkflowExecutionRawHistoryResponse {
+  10: optional binary branchToken
+  20: optional binary nextPageToken
+  30: optional list<DataBlob> historyBatches
+  40: optional map<string, ReplicationInfo> replicationInfo
```

Comment:
I don't like returning part of mutable state along with this call.  ReplicationInfo and EventStoreVersion are part of mutable state and should only be exposed through GetMutableState API.

Reply:
GetMutableState API is not exposed to public side, while the new API providing history events for re-sending message when Kafka fail to deliver message should also include necessary information.

anyway, the new API is moved to admin handler.